compiler/src/dotty/tools/dotc/transform/SymUtils.scala

```diff
@@ -100,6 +100,17 @@ object SymUtils:
         case _ =>
           false
 
+    def isGivenInstanceSummoner(using Context): Boolean =
```

Comment:
The code looks good but I am not sure why given instance summoners should not be reported in semanticdb. Can you explain?

Reply:
Nothing should be reported in semanticdb that doesn't have a real source code position. It can be added to synthetic section, but that will be done later on.

Reply:
then again with a given instance I think the mental model is we are defining a term, not a class

Reply:
After discussion we decided to try and favour the given instance method over the structural given class

Reply:
the issue we have is that
```scala
given [T]: Z[T] with {}
```
generates
```scala
final given def given_Z_T[T]: given_Z_T[T] = new given_Z_T[T]
given class given_Z_T[T] extends Z[T]
```
so it would be correct for occurrences in `: Z[T]` to reference the `T` in the *class* `given_Z_T`, so the only question is what should the definition of the type parameter `given [T]: ` belong to in semanticdb - we could say the method, but that is quite strange visually, as then you get the reference in the return type coming from a different definition

Reply:
commit https://github.com/lampepfl/dotty/pull/12979/commits/7a01b0f52cff19c6a7fe28377a5e65a69ef3531d highlights this difference

Reply:
I would say `*<-givens::InventedNames$package.given_Z_T#[T]` looks better, because givens having the `with` cause look more akin do classes than methods, but on the other hand this is neither method nor class from the code point of view. We could introduce another way of showing givens (we are already discussing a new symbol construct for extension methods), until then in reality both approaches are ok to use here.